IMDA-BTG/aiverify
# Fix/scr frontend

## Description

Add more validations and checks for uploaded files and folders
- Validate all folder names
- Validate mimetype of upload files
- Check that upload filepath is under the "uploads" directory
- Add additional unit test cases to test for invalid uploads

In additional, the package axios under the portal has been updated to 1.6.7. 

## Motivation and Context

To mitigate security risks

## Type of Change













fix: A bug fix 

## How to Test

1. Upload a valid data file and ensure that it can be uploaded
2. Upload a valid data folder with images and ensure that it can be uploaded
3. Upload a data folder with invalid folder name and check that the system response with error.
4. Upload a valid model file and ensure that it can be uploaded

## Checklist

Please check all the boxes that apply to this pull request using "x":


## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
